support/crown/src/unrooted_must_root.rs

```diff
@@ -79,6 +79,26 @@ fn is_unrooted_ty<'tcx>(
                     false
                 } else if has_attr(did.did(), sym.allow_unrooted_interior) {
                     false
+                } else if match_def_path(
+                    cx,
+                    did.did(),
+                    &[sym.js, sym.gc, sym.custom, sym.CustomAutoRooter],
+                ) || match_def_path(
+                    cx,
+                    did.did(),
+                    &[sym.js, sym.rust, sym.custom, sym.CustomAutoRooterGuard],
+                ) || match_def_path(
+                    cx,
+                    did.did(),
+                    &[sym.mozjs_sys, sym.jsgc, sym.Rooted],
+                ) || match_def_path(
+                    cx,
+                    did.did(),
+                    &[sym.mozjs, sym.gc, sym.root, sym.RootedGuard],
+            ) {
+                    // CustomAutoRooter[Guard]/Rooted[Guard] would use #[allow_unrooted_interior]
+                    // if they were not defined in another crate.
```

Comment:
Actually I think add such annotations to mozjs should work. In case no crown is active it should only trigger unknown_lint warning, so let's try it. This way we can actually write tests (I think).

Reply:
Some of the types (eg. Heap) are generated by bindgen, so I don't know if there is a way to apply the annotation there.

Reply:
BTW there is also mechanism for adding attributes to bindgen generated stuff: https://github.com/rust-lang/rust-bindgen/pull/2866

Reply:
I really liked this idea, so I put together an attempt that compiles:
* https://github.com/servo/mozjs/compare/main...jdm:mozjs:crown?expand=1
* https://github.com/servo/servo/compare/main...jdm:servo:mozjs-crow?expand=1